### PR TITLE
AG-9206 - Add Options Reference

### PR DESCRIPTION
**Pull request by taktran**
## Changes

* Add `jsObjectPropertiesView` markdoc component, based on `ExpandableSnippet` component on grid. This splits the `ExpandableSnippet` into 2 columns, with navigation which is selectable on the left (`JsObjectView`) and details of the selected item on the right (`JsObjectDetails`). Both are constructed based on the model from `buildModel`
* Remove `<expandable-snippet>` references in docs
* Create `charts-options` page and use `jsObjectPropertiesView` component
* Update `InterfaceDocumentation` and `ApiDocumentation` to use arrays for `names` and `exclude`. Convert from strings at the astro layer
* Update styles of the navigation in `JsObjectView` to:
  * Remove comments
  * Hide types
  * Use `,` for ending punctuation
  * Use `optional` styles for `?`

## Caveats

* Still need to implement:
  * No selection view ie, display all options
  * Arrays in names eg, `0.axes`, `series.[].0..type` and `1.theme.baseTheme`
  * Empty values between the path ie, the `..` in `series.[].0..type`
  * Some meta items are too long eg, `AgBaseCartesianThemeOptions<AgCartesianSeriesTheme> & T`

### CODE REVIEW

File: packages/ag-charts-website/src/features/api-documentation/components/JsObjectView.module.scss [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    line-height: var(--code-line-height);
+    background-size: 1px calc(1em * var(--code-line-height) * 2);
+}
+
+.jsonProperty {
+    position: relative;
+}
+
+.jsonProperty .jsonProperty {
+    &::before {
```
Comment (markdurrant) [inline packages/ag-charts-website/src/features/api-documentation/components/JsObjectView.module.scss]: Can remove nesting here

File: packages/ag-charts-website/src/features/api-documentation/components/JsObjectDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+interface Props {
+    selection: JsObjectSelection;
+    framework: Framework;
+}
+
+function JsObjectDetailsContainer({ children }: { children: ReactNode }) {
+    return (
+        <div className={styles.apiReferenceOuter}>
+            <table className={classnames(styles.reference, styles.apiReference, 'no-zebra')}>
+                <colgroup>
```
Comment (markdurrant) [inline packages/ag-charts-website/src/features/api-documentation/components/JsObjectDetails.tsx]: What's the purpose of the `<colgroup>` here. Is it an accessibility thing?

File: packages/ag-charts-website/src/features/api-documentation/components/JsObjectView.module.scss (hunk 1 above)
Comment (taktran) [reply packages/ag-charts-website/src/features/api-documentation/components/JsObjectView.module.scss]: 👍🏼  Removed

File: packages/ag-charts-website/src/features/api-documentation/components/JsObjectDetails.tsx (hunk 2 above)
Comment (taktran) [reply packages/ag-charts-website/src/features/api-documentation/components/JsObjectDetails.tsx]: It doesn't seem to be used for anything, so I've removed it.

**Review by markdurrant — APPROVED** [review]
Ship it 🚢

**Pull request merged**